node/pkg/altpub/alternate_pub.go

```diff
@@ -0,0 +1,534 @@
+	}
+
+	baseUrl := fields[1]
+	if valid := common.ValidateURL(baseUrl, []string{"http", "https"}); !valid {
+		return nil, fmt.Errorf("invalid url in --additionalPublishEndpoint '%s': must be `http` or `https`", config)
+	}
+
+	delay := time.Duration(0)
+	if len(fields) > 2 {
+		var err error
+		delay, err = time.ParseDuration(fields[2])
+		if err != nil {
+			return nil, fmt.Errorf("invalid delay duration in --additionalPublishEndpoint '%s', delay %s: %w", config, fields[2], err)
+		}
+	}
+
+	enabledChainsMap := make(map[vaa.ChainID]struct{})
+	exceptFor := false
+	if len(fields) > 3 {
+		str := fields[3]
+		if strings.HasPrefix(str, "-") {
+			exceptFor = true
+			str = str[1:]
+		}
+		chainIds := strings.Split(str, ",")
+		for _, str := range chainIds {
+			chainId, err := vaa.StringToKnownChainID(str)
+			if err != nil {
+				return nil, fmt.Errorf("invalid chain ID --additionalPublishEndpoint '%s' ('%s'): %w", config, str, err)
+			}
+
+			enabledChainsMap[chainId] = struct{}{}
+		}
+	}
+
+	enabledChains := EnabledChains{exceptFor, enabledChainsMap}
+
+	// Only create an observation batch channel if batching is enabled.
+	var obsvBatchChan chan *gossipv1.Observation
+	if delay != 0 {
+		obsvBatchChan = make(chan *gossipv1.Observation, ObservationChanSize)
+	}
+
+	ep := &Endpoint{
+		label:         label,
+		baseUrl:       baseUrl,
+		delay:         Delay(delay),
+		enabledChains: enabledChains,
+		obsvBatchChan: obsvBatchChan,
+	}
+
+	// Format our topic URLs and add the endpoint to the list.
+	ep.createUrls()
+	return ep, nil
+}
+
+// Run is the runnable for the alternate publisher. It creates the various workers and then waits for shutdown.
+func (ap *AlternatePublisher) Run(ctx context.Context) error {
+	errC := make(chan error)
+
+	ap.logger.Info("Starting alternate publisher", zap.Int("numEndpoints", len(ap.endpoints)))
+
+	client, err := ap.createClient()
+	if err != nil {
+		return fmt.Errorf("failed to create http client: %w", err)
+	}
+
+	ap.startHttpWorkers(ctx, client, errC, NumWorkersPerEndpoint*len(ap.endpoints))
+
+	for _, ep := range ap.endpoints {
+		ap.logger.Info("Enabling endpoint", zap.String("endpoint", ep.label), zap.String("url", ep.baseUrl), zap.Stringer("delay", ep.delay), zap.Stringer("enabledChains", ep.enabledChains))
+		if ep.delay != 0 {
+			worker := &BatchWorker{ap, ep}
+			common.RunWithScissors(ctx, errC, fmt.Sprintf("alt_pub_batcher_%s", ep.label), worker.batchWorker)
+		}
+	}
+
+	// Wait until shutdown or an error occurs.
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	case err := <-errC:
+		return err
+	}
+}
+
+// PublishObservation publishes a signed observation to all the endpoints that care about it. It handles both immediate and delayed publishers.
+func (ap *AlternatePublisher) PublishObservation(emitterChain vaa.ChainID, obs *gossipv1.Observation) {
+	var data []byte
+	for _, ep := range ap.endpoints {
+		if ep.shouldPublish(emitterChain) {
```

Comment:
If this was modified to `if !ep.shouldPublish(emitterChain) { continue }`, a level of indentation could be removed for the rest of the code.